### ISSUE TITLE
Posting the way you are using small_gicp

### ISSUE THREAD

**Issue by koide3** [koide3/small_gicp#3]
TL; DR: Sharing that you are using small_gicp (and related packages) helps the author receive recognition within his organization and enables continued work on this project.

Long description: The author of this package is NOT a full-time OSS developer but just a researcher who dedicates the majority of his working hours to other assignments. His time for OSS projects is limited, and to sustain his involvement, internal acknowledgment within his organization is crucial. Metrics like the number of stars and forks are often overlooked by executives, who may not grasp their significance. Conversely, stating "this package is used by A/B universities and C/D corporations" can effectively communicate its value. So, if you find this package useful for your work, we kindly ask you to leave a comment in this thread. Just saying hello to the author is appreciated. If you could share about your organization and how you use the package, that would immensely support the author.

**Comment by AndreasNagel** [comment]
I have not used small_gicp directly, but I used it's predecessor fast_gicp through the hdl_graph_slam package for testing the feasibility of localizing the robot using a tactile sensor array during my master thesis in TalTech university, Tallinn and I believe that same package was used in the robominers project later on by the researchers at TalTech, although that needs confirming. small_gicp is a rather new project, so have not directly used that yet in everyday work. Thank you for your work and sharing it with the world!

**Comment by koide3** [comment]
@AndreasNagel Thanks for sharing your work! It becomes a big help for me!

**Comment by SakodaShintaro** [comment]
Hello. I am one of the developers at TIER IV. At TIER IV, we are developing an open source autonomous driving software, [Autoware](https://github.com/autowarefoundation/autoware), and we are using [ndt_omp](https://github.com/tier4/ndt_omp) in it. It works very stable, and we are grateful that it is easy to make our own changes (such as dynamic map loading and covariance estimation with scores).

Recently, we have been discussing the transition to fast_gicp, and although it is still under consideration, we will also consider this small_gicp. As a first step, (it is a personal work), I have evaluated small_gicp at AWSIM Nishi-Shinjuku and obtained good results. :+1:  <https://tokumini.hatenablog.com/entry/2024/04/07/110000> (Sorry in Japanese)

We are greatly helped by your code. Thank you for publishing such great software!

**Comment by koide3** [comment]
@SakodaShintaro Thank a lot for your comment! It is a pleasure to see my packages have been used in real situations. 

Regarding the evaluation, I noticed that there was a bug that made termination criteria unchanged with the PCL interface. PR https://github.com/koide3/small_gicp/pull/15 would fix this issue and make comparison fair.

**Comment by koide3** [comment]
@grischi Thanks!!

**Comment by hridaybavle** [comment]
Hello @koide3, 

Amazing work again with the small_gicp. We have been using the hdl_graph_slam as a backbone for our [lidar_s_graphs](https://github.com/snt-arg/lidar_s_graphs/) since a while now and the scan matching algorithm algorithm uses fast_gicp. I think adding small_gicp to the scan matching will further improve the performance. Thanks again for the amazing work!!

**Comment by koide3** [comment]
Hi Hriday, Thanks a lot for your kind comment! I hope we can meet again at some conference :)

**Comment by KOKIAOKI** [comment]
Hello @koide3! 
Thank you for providing sophisticated open source code!
Currently I fully replaced pcl::PointCloud with small_gicp::PoincCloud in my point cloud registration code because it allows us to manage the points and covariance at the same time and be parallelized everywhere. Furthermore, thanks to this clear c++ implementation, I understood the simple optimization way in point cloud registration. Regardless of whether this source code is maintained, I will refer to this source code continuously in order to be readable code and enhance point cloud registration.

small_gicp must be the amazing gift for the robotics researcher and developer. I will strongly recommend small_gicp to my coworker and friends.
Thank you again!

**Comment by koide3** [comment]
Thanks @KOKIAOKI !

**Comment by RintoDeVries** [comment]
Hi! We are a drone inspection company using small_gicp (and previously fast_gicp) both for lidar odometry as well as icp localization. Thanks for the amazing work!

**Comment by koide3** [comment]
Thank you @RintoDeVries !

**Comment by bpoebiapl** [comment]
Hello. I'm a researcher in whereable.ai, an autonomous driving startup in Korea. Your work is amazing and we expect that applying your results to the SLAM framework of our system will lead to significant performance improvements. The relevant industry will make great progress through your results. We would like to thank not only you but also AIST for their support in achieving these results.

**Comment by koide3** [comment]
Thank you for your comment @bpoebiapl !

**Comment by kennyjchen** [comment]
Hi @koide3 -- your work has been vital for my PhD research and beyond, including [DLO](https://github.com/vectr-ucla/direct_lidar_odometry) and [DLIO](https://github.com/vectr-ucla/direct_lidar_inertial_odometry). Your hard work and commitment to OSS is deeply appreciated, and my sincerest thanks to you for your contributions to this community!

**Comment by koide3** [comment]
Thanks @kennyjchen ! I've been following your works, and I'm so glad to see that my package could help you great jobs!

**Comment by narutojxl** [comment]
Hi Dr. @koide3, I am a robot practitioner. Since of year from 2019 to now, i have learned a lot knowledge from your series excellent works, including hdl_localization, fast_gicp, interactive_slam etc,  discussing with you about these  repos issues on github. I'm very  appreciated, and want to give my sincerest thanks to you.  Also thanks Dr. @kennyjchen, i have  also learned a lot  knowledge from your DLO and DLIO work. I'm much appreciated.

**Comment by aserbremen** [comment]
Dear @koide3, thanks for this new contribution to the robotics community. Today, I performed a proof of concept for integrating [small_gicp](https://github.com/koide3/small_gicp) into [mrg_slam](https://github.com/aserbremen/Multi-Robot-Graph-SLAM), the multi-robot SLAM based on [hdl_graph_slam](https://github.com/koide3/hdl_graph_slam). 

I ran scan_matching_odometry with fast_gicp and small_gicp using 8 threads and ~4,700 points in the point cloud. fast_gicp took ~5.6 ms per scan and small_gicp ~3.2 ms, which also aligns with with the expected performance increase, which is great.  The benefit in the loop detector was not as big but still noticable. 

This post might be a little to technical. But could you take a look on how I register small_gicp in mrg_slam [here](https://github.com/aserbremen/mrg_slam/blob/dba87fa910ef66e9b3afe529ed4d280cdab4ac13/src/mrg_slam/registrations.cpp#L46) and let me know if I should set more registrations options. The following are the default values: 

    reg_num_threads: 8
    reg_transformation_epsilon: 0.1
    reg_maximum_iterations: 64
    reg_max_correspondence_distance: 2.0
    reg_correspondence_randomness: 20

**Comment by koide3** [comment]
@aserbremen 
I'm so happy to see that small_gicp has been integrated into mrg_slam! Thanks a lot for letting me know that! The setup looks good; small_gicp can be used as just a drop-in replacement for fast_gicp ;)

**Comment by JACKLiuDay** [comment]
@koide3 Hello, thank you very much for your open-source work. I am from the Robotics Center at Shandong University, China. We have successfully used Fast VGICP to implement autonomous navigation on the quadruped robot we are developing, and have explored the application of legged robots in power line inspection. We will continue to follow the progress of Small gicp and stabilize our existing autonomous navigation framework.

**Comment by GyoungTaeChae** [comment]
Thank you for always sharing such great resources. I really appreciate your work! I 'm goint to study ICP by analyzing this code, and it seems very helpful. I hope you keep up the great work in long time!

**Comment by LimHyungTae** [comment]
Dear @koide3 , I always appreciate you sharing your wonderful codes. Its quality is wonderful and just works well in most cases, so I also leverage some codes in my [KISS-Matcher project](https://github.com/MIT-SPARK/KISS-Matcher).

**Comment by thomasemter** [comment]
Dear @koide3,

We make use of small_gicp in a package for [multisensor calibration](https://github.com/FraunhoferIOSB/multisensor_calibration). Thank you for your wonderful work and making it open source.

**Comment by ga58lar** [comment]
Dear @koide3,
Thanks a lot for your excellent open-source contributions!
I was already using fast_gicp and the hdl_graph_slam during my studies and am now again using this library and others of yours for my zero-drift SLAM [OpenLiDARMap](https://github.com/TUMFTM/Openlidarmap).

**Comment by kennyjchen** [comment]
Hi @koide3 -- just wanted to say thanks again for this amazing work. We've recently transitioned from `fast_gicp` to `small_gicp`, and the performance gains are quite impressive 😄 Huge thanks to you again for the excellent package! It's very well written and well documented.

**Comment by tsengapola** [comment]
Hi @koide3, we were looking for a high-efficiency voxelization library, and after integrating yours into our 3D navigation stack, [dddmr_navigation](https://github.com/dfl-rlab/dddmr_navigation), we saw a 20% speed improvement. Thank you for your great contribution!
